src/imap.rs

```diff
@@ -1411,19 +1418,27 @@ impl Imap {
                         err
                     );
                 }
-            };
-
-            while let Some(Ok(msg)) = msgs.next().await {
+            }
+            .filter_map(|r| async {
+                match r {
+                    Ok(msg) => Some(msg),
+                    Err(_) => None,
+                }
+            })
+            .collect::<Vec<_>>()
+            .await;
+            msgs.sort_unstable_by(|a, b| {
+                uid_indices
+                    .get(&a.uid.unwrap_or_default())
+                    .cmp(&uid_indices.get(&b.uid.unwrap_or_default()))
+            });
+
+            for msg in msgs {
                 let server_uid = msg.uid.unwrap_or_default();
 
-                if !server_uids.contains(&server_uid) {
-                    warn!(
-                        context,
-                        "Got unwanted uid {} not in {:?}, requested {:?}",
-                        &server_uid,
-                        server_uids,
```

Comment:
Removed logging of `server_uids` Vec here. Better to log huge structs only once if some error happens

Reply:
Makes sense, even though i can't remember this special line here cluttering the log.

Since AFAIK we never had the problem that we generated the wrong uid set (either me or @link2xt would know), it would even be fine to not do the effort (+ code complexity) with the `count_extra` variable, and just remove the logging here.

Reply:
Ok, removed this log and added your comment instead